server/storage/mvcc/watcher.go

```diff
@@ -134,7 +137,10 @@ func (ws *watchStream) Watch(id WatchID, key, end []byte, startRev int64, fcs ..
 
 	w, c := ws.watchable.watch(key, end, startRev, id, ws.ch, fcs...)
 
-	ws.cancels[id] = c
+	ws.cancels[id] = func() {
+		defer trace.SpanFromContext(ctx).End()
```

Comment:
I think we should call trace.SpanFromContext(ctx) out of func() {} first.

Reply:
Changed to:
```
	span := trace.SpanFromContext(ctx)
	ws.cancels[id] = func() {
		defer span.End()
		c()
	}
```